Submitted By:            Igor Živković <contact@igor-zivkovic.from.hr>
Date:                    2016-08-17
Initial Package Version: 1.25.0
Upstream Status:         Already in upstream patch repo
Origin:                  Upstream
Description:             This patch contains bug fixes identified upstream.

--- busybox-1.25.0.orig/archival/gzip.c
+++ busybox-1.25.0/archival/gzip.c
@@ -2220,10 +2220,7 @@
 	opt >>= ENABLE_GUNZIP ? 7 : 5; /* drop cfv[dt]qn bits */
 	if (opt == 0)
 		opt = 1 << 6; /* default: 6 */
-	/* Map 1..3 to 4 */
-	if (opt & 0x7)
-		opt |= 1 << 4;
-	opt = ffs(opt >> 3);
+	opt = ffs(opt >> 4); /* Maps -1..-4 to [0], -5 to [1] ... -9 to [5] */
 	max_chain_length = 1 << gzip_level_config[opt].chain_shift;
 	good_match	 = gzip_level_config[opt].good;
 	max_lazy_match	 = gzip_level_config[opt].lazy2 * 2;
--- busybox-1.25.0.orig/networking/libiproute/iproute.c
+++ busybox-1.25.0/networking/libiproute/iproute.c
@@ -362,10 +362,9 @@
 		req.r.rtm_scope = RT_SCOPE_NOWHERE;
 
 	if (cmd != RTM_DELROUTE) {
+		req.r.rtm_scope = RT_SCOPE_UNIVERSE;
 		if (RTPROT_BOOT != 0)
 			req.r.rtm_protocol = RTPROT_BOOT;
-		if (RT_SCOPE_UNIVERSE != 0)
-			req.r.rtm_scope = RT_SCOPE_UNIVERSE;
 		if (RTN_UNICAST != 0)
 			req.r.rtm_type = RTN_UNICAST;
 	}
--- busybox-1.25.0.orig/networking/ntpd.c
+++ busybox-1.25.0/networking/ntpd.c
@@ -2051,6 +2051,13 @@
 		goto bail;
 	}
 
+	/* Respond only to client and symmetric active packets */
+	if ((msg.m_status & MODE_MASK) != MODE_CLIENT
+	 && (msg.m_status & MODE_MASK) != MODE_SYM_ACT
+	) {
+		goto bail;
+	}
+
 	query_status = msg.m_status;
 	query_xmttime = msg.m_xmttime;
 
